File: backend/app/schemas/flashcard.py
from pydantic import BaseModel
from typing import List, Dict

class FlashcardResponse(BaseModel):
    flashcard_id: int
    word: str
    translation: str
    type: str
    english_equivalents: List[str]
    definition: str
    english_definition: str
    example_sentence: str
    english_sentence: str
    category: str
    lesson_id: int
    options: List[Dict[str, str]]

File: frontend/src/types/flashcard.ts
export interface Flashcard {
  flashcard_id: number;
  word: string;
  translation: string;
  type: string;
  english_equivalents: string[];
  definition: string;
  english_definition: string;
  example_sentence: string;
  english_sentence: string;
  category: string;
  lesson_id: number;
  options: { id: string; option_text: string }[];
}